src/main/java/hudson/plugins/ec2/AmazonEC2Cloud.java

```diff
@@ -65,6 +60,8 @@
      */
     private String region;
 
+    private String altEC2Endpoint;
```

Comment:
Is the endpoint not required for the AmazonEC2 Client object? I don't see it specified for the object used when spinning up instances etc.

Reply:
I didn't change any of the existing code that checks for endpoint. I will look at switching the getEc2EndpointUrl logic to better handle where it's directly set and not use the region if the endpoint URL is configured. I think I did see a ticket where the region -> endpoint logic didn't work in C2S, but I won't have the ability to fully test that.